### PR TITLE
V2 Donations

### PR DESCRIPTION
**Pull request by 0xKurt**
- Ensure you are on allo-v2
- Donate to multiple rounds using ETH 
- Donate to multiple rounds using DAI (please ensure you mint it using address below )
- Donate to multiple rounds using USDC (please ensure you mint it using address below )

Please use test rounds : 
https://explorer-staging-git-v2-donations-explorer-grants-stack.vercel.app/#/round/11155111/90
https://explorer-staging-git-v2-donations-explorer-grants-stack.vercel.app/#/round/11155111/91
https://explorer-staging-git-v2-donations-explorer-grants-stack.vercel.app/#/round/11155111/92

You can mint DAI and USDC here:
DAI:
https://sepolia.etherscan.io/address/0xa9dd7983B57E1865024d27110bAB098B66087e8F#writeContract#F4
USDC:
https://sepolia.etherscan.io/address/0x78e0D07C4A08adFfe610113310163b40E7e47e81#writeContract#F4

Fixes [#2938](https://github.com/gitcoinco/grants-stack/issues/2938)

### CODE REVIEW

**Review by thelostone-mc — APPROVED** [review]

File: packages/common/src/allo/backends/allo-v2.ts [hunk 1]
```diff
@@ -84,26 +87,87 @@ export class AlloV2 implements Allo {
...
 
-  async voteUsingMRCContract(
-    _publicClient: PublicClient,
-    _chainId: ChainId,
-    _token: VotingToken,
-    _groupedVotes: Record<string, Hex[]>,
-    _groupedAmounts: Record<string, bigint>,
-    _nativeTokenAmount: bigint,
-    _permit?: {
+  async donate(
```
Comment (boudra) [inline packages/common/src/allo/backends/allo-v2.ts:90]: 🙏

File: packages/common/src/allo/backends/allo-v2.ts [hunk 2]
```diff
@@ -84,26 +87,87 @@ export class AlloV2 implements Allo {
...
+            BigInt(permit.deadline ?? Number.MAX_SAFE_INTEGER),
+            permit.sig.v,
+            permit.sig.r as Hex,
+            permit.sig.s as Hex,
+          ],
+        });
+      }
+    } else {
+      /* Tried voting using erc-20 but no permit signature provided */
+      throw new Error(
```
Comment (boudra) [inline packages/common/src/allo/backends/allo-v2.ts:161]:
```suggestion
      throw new AlloError(
```

**Review by boudra — APPROVED** [review]

**Review by michellema1208 — CHANGES_REQUESTED** [review]
@0xKurt / @thelostone-mc / @codenamejason - I'm having trouble adding different projects across different rounds to the same cart -- it seems like adding a project in one round to a cart triggers adding a different project in a different round to cart as well, and also the projects don't appear in the same cart. Same things happens when I try to add the same project across different rounds to the same cart. 

Expected behavior: I am able to add different projects from different rounds to the same cart on v2. If I add the same project from two different rounds to the same cart, I see the same project in my cart twice, once in each round. 

Loom: https://www.loom.com/share/4891940b8095410a83295a7b9c767c68?focus_title=1&muted=1&from_recorder=1

**Review by michellema1208 — APPROVED** [review]
LGTM - great work Kurt and team!

**Pull request merged**